sdk/ml/azure-ai-ml/azure/ai/ml/entities/_job/automl/automl_job.py

```diff
@@ -48,13 +69,25 @@ def __init__(
     ) -> None:
         """Initialize an AutoML job entity.
 
-        :param task_details: The task configuration of the job. This can be Classification, Regression, etc.
-        :param resources: Resource configuration for the job.
-        :param identity: Identity that training job will use while running on compute.
-        :type identity: Union[ManagedIdentity, AmlToken, UserIdentity]
-        :param queue_settings: Queue settings for the job.
-        :type queue_settings: QueueSettings
-        :param kwargs:
+        Constructor for an AutoMLJob.
+
+        :keyword resources: Resource configuration for the AutoML job, defaults to None
+        :paramtype resources: typing.Optional[ResourceConfiguration]
+        :keyword identity: Identity that training job will use while running on compute, defaults to None
+        :paramtype identity: typing.Optional[ typing.Union[ManagedIdentityConfiguration, AmlTokenConfiguration
+            , UserIdentityConfiguration] ]
+        :keyword environment_id: The environment id for the AutoML job, defaults to None
```

Comment:
these 3 are not parameters of init ..

Reply:
Those 3 parameters are being retrieve from the kwargs. Here is the code copied from the init method: 

`

        kwargs[TYPE] = JobType.AUTOML
        self.environment_id = kwargs.pop("environment_id", None)
        self.environment_variables = kwargs.pop("environment_variables", None)
        self.outputs = kwargs.pop("outputs", None)

        super().__init__(**kwargs)
`

Reply:
in this case what is the drawback if we explicitly add them?